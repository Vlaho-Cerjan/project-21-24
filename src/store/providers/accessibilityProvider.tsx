import { ThemeProvider } from "@emotion/react";
import { Theme, createTheme } from "@mui/material";
import { createContext, useEffect, useState } from "react";
import { useAppDispatch, useAppSelector } from "../../hooks";
import darkThemeOptions from "../../styles/theme/darkThemeOptions";
import lightThemeOptions from "../../styles/theme/lightThemeOptions";
import { AccessibilityState, setFontSize, setLanguage, setMode } from "../slices/accessibilitySlice";
import { fetchEnum } from "../slices/enumSlice";

export const defaultLocale = "en";

const light = createTheme(lightThemeOptions);
const dark = createTheme(darkThemeOptions);

const Themes = {
    light,
    dark,
}

export const AccessibilityContext = createContext<{
    accessibility: AccessibilityState,
    theme: Theme,
}>({
    accessibility: {
        fontSize: 10,
        language: 'en',
        mode: 'light',
        isDark: false,
    },
    theme: light
})


export const AccessibilityProvider = ({ children }: { children: React.ReactNode }) => {
    const { fontSize, language, mode, isDark } = useAppSelector(state => state.accessibility);
    const dispatch = useAppDispatch();
    const [theme, setTheme] = useState<Theme>(Themes[mode]);

    useEffect(() => {
        const storageAcc = localStorage.getItem('accessibility');
        if (storageAcc) {
            const parsedAcc: AccessibilityState = JSON.parse(storageAcc)
            dispatch(setFontSize(parsedAcc.fontSize))
            dispatch(setLanguage(parsedAcc.language))
            dispatch(setMode(parsedAcc.mode))

            if (parsedAcc.mode) {
                setTheme(Themes[parsedAcc.mode]);
            }
        }

        if(dispatch) dispatch(fetchEnum());
    }, [dispatch])

    useEffect(() => {
        if (fontSize) {
            const fontSizeClass = new RegExp(/\bfontSize_.+?\b/, 'g')

            if (document.body.className.match(fontSizeClass)) {
                document.body.className = document.body.className.replace(fontSizeClass, '')
            }
            document.body.classList.add('fontSize_' + fontSize);
        }
    }, [fontSize])

    useEffect(() => {
        if (mode) setTheme(Themes[mode]);

        if (isDark && !document.body.classList.contains("dark")) document.body.classList.add("dark");
        else if (!isDark && document.body.classList.contains("dark")) document.body.classList.remove("dark");
    }, [mode]);

    const contextValue = {
        accessibility: {
            fontSize,
            language,
            mode,
            isDark,
        },
        theme
    }

    return (
        <AccessibilityContext.Provider value={contextValue}>
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </AccessibilityContext.Provider>
    )
}