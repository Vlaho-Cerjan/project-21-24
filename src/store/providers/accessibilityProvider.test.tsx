// accessibility provider test

import React from "react";
import { AccessibilityContext } from "./accessibilityProvider";
import { act, getByTestId, render } from "test/utilities";
import { configStore } from "..";
import { describe, it } from 'vitest';

const TestingComponent = () => {
    const { accessibility, theme } = React.useContext(AccessibilityContext);

    return (
        <>
            <div data-testid="accessibility">{JSON.stringify(accessibility)}</div>
            <div data-testid="theme">{JSON.stringify(theme)}</div>
        </>
    )
}

describe('AccessibilityProvider', () => {
    it('should render correctly', () => {
        render(<TestingComponent />);

        expect(document.body).toMatchSnapshot();
    });

    it('should render with localStorage accessibility already set', () => {
        localStorage.setItem('accessibility', JSON.stringify({
            fontSize: 20,
            language: 'fr',
            mode: 'dark',
            isDark: true,
        }));

        render(<TestingComponent />);

        const accessibilityText = getByTestId(document.body, 'accessibility');
        const themeText = getByTestId(document.body, 'theme');

        expect(accessibilityText).toHaveTextContent(JSON.stringify({
            fontSize: 20,
            language: 'fr',
            mode: 'dark',
            isDark: true,
        }));

        const theme = JSON.parse(themeText.textContent || '');

        expect(theme.palette.mode).toBe('dark');
    });

    it('should render correctly with dark mode and fontSize and language and isDark', () => {
        const store = configStore;

        render(<TestingComponent />);

        act(() => {
            store.dispatch({ type: "accessibility/setMode", payload: "dark" });
            store.dispatch({ type: "accessibility/setFontSize", payload: 20 });
            store.dispatch({ type: "accessibility/setLanguage", payload: "fr" });
            store.dispatch({ type: "accessibility/setIsDark", payload: true });
        });

        const accessibilityText = getByTestId(document.body, 'accessibility');
        const themeText = getByTestId(document.body, 'theme');

        expect(accessibilityText).toHaveTextContent(JSON.stringify({
            fontSize: 20,
            language: 'fr',
            mode: 'dark',
            isDark: true,
        }));

        const theme = JSON.parse(themeText.textContent || '');

        expect(theme.palette.mode).toBe('dark');
    });

    it('should render correctly with light mode and fontSize and language and isDark', () => {
        const store = configStore;

        render(<TestingComponent />);

        const accessibilityText = getByTestId(document.body, 'accessibility');
        const themeText = getByTestId(document.body, 'theme');

        act(() => {
            store.dispatch({ type: "accessibility/setMode", payload: "light" });
            store.dispatch({ type: "accessibility/setFontSize", payload: 20 });
            store.dispatch({ type: "accessibility/setLanguage", payload: "fr" });
            store.dispatch({ type: "accessibility/setIsDark", payload: true });
        });

        expect(accessibilityText).toHaveTextContent(JSON.stringify({
            fontSize: 20,
            language: 'fr',
            mode: 'light',
            isDark: false,
        }));

        const theme = JSON.parse(themeText.textContent || '');

        expect(theme.palette.mode).toBe('light');
    });
});