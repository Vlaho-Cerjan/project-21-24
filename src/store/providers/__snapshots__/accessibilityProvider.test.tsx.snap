// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccessibilityProvider > should render correctly 1`] = `
<body
  class="fontSize_10"
>
  <div>
    <div
      data-testid="accessibility"
    >
      {"fontSize":10,"language":"en","mode":"light","isDark":false}
    </div>
    <div
      data-testid="theme"
    >
      {"breakpoints":{"keys":["xs","sm","md","lg","xl"],"values":{"xs":0,"sm":600,"md":900,"lg":1200,"xl":1536},"unit":"px"},"direction":"ltr","components":{},"palette":{"mode":"light","text":{"primary":"rgba(0,0,32,0.76)","secondary":"rgba(0,0,32,0.48)","disabled":"rgba(0, 0, 0, 0.38)"},"primary":{"main":"#288CF0","dark":"#2B7AE2","light":"rgb(83, 163, 243)","contrastText":"#fff"},"success":{"main":"#8CD228","dark":"#7CB822","contrastText":"#fff","light":"rgb(163, 219, 83)"},"error":{"main":"#EE5A5A","light":"rgb(241, 123, 123)","dark":"rgb(166, 62, 62)","contrastText":"#fff"},"background":{"default":"#f8f8fa","paper":"#fff"},"divider":"rgba(0,0,32,0.04)","common":{"black":"#000","white":"#fff"},"secondary":{"main":"#9c27b0","light":"#ba68c8","dark":"#7b1fa2","contrastText":"#fff"},"warning":{"main":"#ed6c02","light":"#ff9800","dark":"#e65100","contrastText":"#fff"},"info":{"main":"#0288d1","light":"#03a9f4","dark":"#01579b","contrastText":"#fff"},"grey":{"50":"#fafafa","100":"#f5f5f5","200":"#eeeeee","300":"#e0e0e0","400":"#bdbdbd","500":"#9e9e9e","600":"#757575","700":"#616161","800":"#424242","900":"#212121","A100":"#f5f5f5","A200":"#eeeeee","A400":"#bdbdbd","A700":"#616161"},"contrastThreshold":3,"tonalOffset":0.2,"action":{"active":"rgba(0, 0, 0, 0.54)","hover":"rgba(0, 0, 0, 0.04)","hoverOpacity":0.04,"selected":"rgba(0, 0, 0, 0.08)","selectedOpacity":0.08,"disabled":"rgba(0, 0, 0, 0.26)","disabledBackground":"rgba(0, 0, 0, 0.12)","disabledOpacity":0.38,"focus":"rgba(0, 0, 0, 0.12)","focusOpacity":0.12,"activatedOpacity":0.12}},"shape":{"borderRadius":12},"unstable_sxConfig":{"border":{"themeKey":"borders"},"borderTop":{"themeKey":"borders"},"borderRight":{"themeKey":"borders"},"borderBottom":{"themeKey":"borders"},"borderLeft":{"themeKey":"borders"},"borderColor":{"themeKey":"palette"},"borderTopColor":{"themeKey":"palette"},"borderRightColor":{"themeKey":"palette"},"borderBottomColor":{"themeKey":"palette"},"borderLeftColor":{"themeKey":"palette"},"borderRadius":{"themeKey":"shape.borderRadius"},"color":{"themeKey":"palette"},"bgcolor":{"themeKey":"palette","cssProperty":"backgroundColor"},"backgroundColor":{"themeKey":"palette"},"p":{},"pt":{},"pr":{},"pb":{},"pl":{},"px":{},"py":{},"padding":{},"paddingTop":{},"paddingRight":{},"paddingBottom":{},"paddingLeft":{},"paddingX":{},"paddingY":{},"paddingInline":{},"paddingInlineStart":{},"paddingInlineEnd":{},"paddingBlock":{},"paddingBlockStart":{},"paddingBlockEnd":{},"m":{},"mt":{},"mr":{},"mb":{},"ml":{},"mx":{},"my":{},"margin":{},"marginTop":{},"marginRight":{},"marginBottom":{},"marginLeft":{},"marginX":{},"marginY":{},"marginInline":{},"marginInlineStart":{},"marginInlineEnd":{},"marginBlock":{},"marginBlockStart":{},"marginBlockEnd":{},"displayPrint":{"cssProperty":false},"display":{},"overflow":{},"textOverflow":{},"visibility":{},"whiteSpace":{},"flexBasis":{},"flexDirection":{},"flexWrap":{},"justifyContent":{},"alignItems":{},"alignContent":{},"order":{},"flex":{},"flexGrow":{},"flexShrink":{},"alignSelf":{},"justifyItems":{},"justifySelf":{},"gap":{},"rowGap":{},"columnGap":{},"gridColumn":{},"gridRow":{},"gridAutoFlow":{},"gridAutoColumns":{},"gridAutoRows":{},"gridTemplateColumns":{},"gridTemplateRows":{},"gridTemplateAreas":{},"gridArea":{},"position":{},"zIndex":{"themeKey":"zIndex"},"top":{},"right":{},"bottom":{},"left":{},"boxShadow":{"themeKey":"shadows"},"width":{},"maxWidth":{},"minWidth":{},"height":{},"maxHeight":{},"minHeight":{},"boxSizing":{},"fontFamily":{"themeKey":"typography"},"fontSize":{"themeKey":"typography"},"fontStyle":{"themeKey":"typography"},"fontWeight":{"themeKey":"typography"},"letterSpacing":{},"textTransform":{},"lineHeight":{},"textAlign":{},"typography":{"cssProperty":false,"themeKey":"typography"}},"mixins":{"toolbar":{"minHeight":56,"@media (min-width:0px)":{"@media (orientation: landscape)":{"minHeight":48}},"@media (min-width:600px)":{"minHeight":64}}},"shadows":["none","0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)","0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)","0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)","0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)","0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)","0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)","0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)","0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)","0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)","0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)","0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)","0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)","0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)","0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)","0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)","0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)","0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)","0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)","0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)","0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)","0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)","0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)","0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)","0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)"],"typography":{"htmlFontSize":16,"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontSize":14,"fontWeightLight":300,"fontWeightRegular":400,"fontWeightMedium":500,"fontWeightBold":700,"h1":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":300,"fontSize":"6rem","lineHeight":1.167,"letterSpacing":"-0.01562em"},"h2":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":300,"fontSize":"3.75rem","lineHeight":1.2,"letterSpacing":"-0.00833em"},"h3":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"3rem","lineHeight":1.167,"letterSpacing":"0em"},"h4":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"2.125rem","lineHeight":1.235,"letterSpacing":"0.00735em"},"h5":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"1.5rem","lineHeight":1.334,"letterSpacing":"0em"},"h6":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":500,"fontSize":"1.25rem","lineHeight":1.6,"letterSpacing":"0.0075em"},"subtitle1":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"1rem","lineHeight":1.75,"letterSpacing":"0.00938em"},"subtitle2":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":500,"fontSize":"0.875rem","lineHeight":1.57,"letterSpacing":"0.00714em"},"body1":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"1rem","lineHeight":1.5,"letterSpacing":"0.00938em"},"body2":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"0.875rem","lineHeight":1.43,"letterSpacing":"0.01071em"},"button":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":500,"fontSize":"0.875rem","lineHeight":1.75,"letterSpacing":"0.02857em","textTransform":"uppercase"},"caption":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"0.75rem","lineHeight":1.66,"letterSpacing":"0.03333em"},"overline":{"fontFamily":"\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif","fontWeight":400,"fontSize":"0.75rem","lineHeight":2.66,"letterSpacing":"0.08333em","textTransform":"uppercase"},"inherit":{"fontFamily":"inherit","fontWeight":"inherit","fontSize":"inherit","lineHeight":"inherit","letterSpacing":"inherit"}},"transitions":{"easing":{"easeInOut":"cubic-bezier(0.4, 0, 0.2, 1)","easeOut":"cubic-bezier(0.0, 0, 0.2, 1)","easeIn":"cubic-bezier(0.4, 0, 1, 1)","sharp":"cubic-bezier(0.4, 0, 0.6, 1)"},"duration":{"shortest":150,"shorter":200,"short":250,"standard":300,"complex":375,"enteringScreen":225,"leavingScreen":195}},"zIndex":{"mobileStepper":1000,"fab":1050,"speedDial":1050,"appBar":1100,"drawer":1200,"modal":1300,"snackbar":1400,"tooltip":1500}}
    </div>
    <div
      class="MuiBox-root css-1puvgnd"
    >
      <div
        class="MuiBox-root css-iah4nu"
      >
        <svg
          height="200px"
          preserveAspectRatio="xMidYMid"
          style="margin: auto; background: none; display: block; shape-rendering: auto;"
          viewBox="0 0 100 100"
          width="200px"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M6.999-2.173c-0.21-1.843-1.216-3.542-2.21-5.061C4.291-7.995,3.814-8.771,3.242-9.477 c-0.389-0.479-0.799-0.955-0.985-1.554c-0.13-0.42,0.033-1.539-0.704-1.467c-0.359,0.035-0.418,0.337-0.471,0.621 c-0.072,0.38,0,0.842,0,1.227c0,4.1,0,8.199,0,12.299c0,1.247,0,2.495,0,3.742c-2.884-1.873-8.377,1.23-8.197,4.705 c0.091,1.754,1.807,2.481,3.362,2.395c1.974-0.108,3.983-1.249,5.113-2.868c1-1.434,0.769-3.163,0.769-4.822 c0-2.398,0-4.797,0-7.195c0-1.72,0-3.439,0-5.159c1.769,2.454,4.472,4.668,4.002,8.037C6.038,1.149,5.681,1.832,5.665,2.494 C5.639,3.53,6.145,2.799,6.343,2.349C6.99,0.875,7.253-0.575,6.999-2.173z"
            fill="#000000"
            transform="translate(30,130)"
          >
            <animatetransform
              attributeName="transform"
              begin="-0.8928571428571428s"
              calcMode="spline"
              dur="1.7857142857142856s"
              keySplines="0 0 1 1;0 0.5 0.5 1;0.5 0 1 0.5;0 0 1 1"
              keyTimes="0;0;0.2;0.4;1"
              repeatCount="indefinite"
              type="translate"
              values="30 130;30 130;30 50;30 130;30 130;"
            />
          </path>
          <path
            d="M6.999-2.173c-0.21-1.843-1.216-3.542-2.21-5.061C4.291-7.995,3.814-8.771,3.242-9.477 c-0.389-0.479-0.799-0.955-0.985-1.554c-0.13-0.42,0.033-1.539-0.704-1.467c-0.359,0.035-0.418,0.337-0.471,0.621 c-0.072,0.38,0,0.842,0,1.227c0,4.1,0,8.199,0,12.299c0,1.247,0,2.495,0,3.742c-2.884-1.873-8.377,1.23-8.197,4.705 c0.091,1.754,1.807,2.481,3.362,2.395c1.974-0.108,3.983-1.249,5.113-2.868c1-1.434,0.769-3.163,0.769-4.822 c0-2.398,0-4.797,0-7.195c0-1.72,0-3.439,0-5.159c1.769,2.454,4.472,4.668,4.002,8.037C6.038,1.149,5.681,1.832,5.665,2.494 C5.639,3.53,6.145,2.799,6.343,2.349C6.99,0.875,7.253-0.575,6.999-2.173z"
            fill="#fccc12"
            transform="translate(70,20)"
          >
            <animatetransform
              attributeName="transform"
              begin="-0.8928571428571428s"
              calcMode="spline"
              dur="1.7857142857142856s"
              keySplines="0 0 1 1;0 0.5 0.5 1;0.5 0 1 0.5;0 0 1 1"
              keyTimes="0;0.1;0.4;0.7;1"
              repeatCount="indefinite"
              type="translate"
              values="70 130;70 130;70 20;70 130;70 130;"
            />
          </path>
          <path
            d="M10.988-11.314c-0.201-0.914-2.865,0.152-3.31,0.255c-2.865,0.661-5.73,1.322-8.594,1.983 C-1.873-8.855-2.84-8.658-3.791-8.413c-0.481,0.124-0.664,0.217-0.719,0.77c-0.18,1.819,0,3.786,0,5.621c0,2.663,0,5.325,0,7.988 c-2.146-1.278-5.411,0.577-6.269,2.683c-1.074,2.636,1.925,3.462,3.907,2.736c1.954-0.715,3.253-2.269,3.343-4.333 c0.046-1.037,0-2.087,0-3.125c0-1.638,0-3.276,0-4.914c0-0.673,0-1.346,0-2.019c0-0.215-0.116-0.848,0-1.036 c0.17-0.277,1.196-0.348,1.535-0.427c1.425-0.329,2.849-0.657,4.274-0.986c2.579-0.595,5.158-1.19,7.737-1.786 c0,3.284,0,6.569,0,9.853c-1.64-0.977-3.949-0.013-5.2,1.179c-0.958,0.913-1.87,2.579-0.84,3.778 c1.198,1.393,3.567,0.71,4.866-0.135C10.164,6.578,10.936,5.347,11,3.788c0.148-3.605,0-7.245,0-10.853c0-0.415,0-0.829,0-1.244 C11-9.247,11.189-10.396,10.988-11.314z"
            fill="#000000"
            transform="translate(40,60)"
          >
            <animatetransform
              attributeName="transform"
              begin="-0.8928571428571428s"
              calcMode="spline"
              dur="1.7857142857142856s"
              keySplines="0 0 1 1;0 0.5 0.5 1;0.5 0 1 0.5;0 0 1 1"
              keyTimes="0;0.3;0.6;0.9;1"
              repeatCount="indefinite"
              type="translate"
              values="40 130;40 130;40 60;40 130;40 130;"
            />
          </path>
          <path
            d="M6.999-2.173c-0.21-1.843-1.216-3.542-2.21-5.061C4.291-7.995,3.814-8.771,3.242-9.477 c-0.389-0.479-0.799-0.955-0.985-1.554c-0.13-0.42,0.033-1.539-0.704-1.467c-0.359,0.035-0.418,0.337-0.471,0.621 c-0.072,0.38,0,0.842,0,1.227c0,4.1,0,8.199,0,12.299c0,1.247,0,2.495,0,3.742c-2.884-1.873-8.377,1.23-8.197,4.705 c0.091,1.754,1.807,2.481,3.362,2.395c1.974-0.108,3.983-1.249,5.113-2.868c1-1.434,0.769-3.163,0.769-4.822 c0-2.398,0-4.797,0-7.195c0-1.72,0-3.439,0-5.159c1.769,2.454,4.472,4.668,4.002,8.037C6.038,1.149,5.681,1.832,5.665,2.494 C5.639,3.53,6.145,2.799,6.343,2.349C6.99,0.875,7.253-0.575,6.999-2.173z"
            fill="#000000"
            transform="translate(60,130)"
          >
            <animatetransform
              attributeName="transform"
              begin="-0.8928571428571428s"
              calcMode="spline"
              dur="1.7857142857142856s"
              keySplines="0 0 1 1;0 0.5 0.5 1;0.5 0 1 0.5;0 0 1 1"
              keyTimes="0;0.5;0.75;1;1"
              repeatCount="indefinite"
              type="translate"
              values="60 130;60 130;60 70;60 130;60 130;"
            />
          </path>
          <path
            d="M10.988-11.314c-0.201-0.914-2.865,0.152-3.31,0.255c-2.865,0.661-5.73,1.322-8.594,1.983 C-1.873-8.855-2.84-8.658-3.791-8.413c-0.481,0.124-0.664,0.217-0.719,0.77c-0.18,1.819,0,3.786,0,5.621c0,2.663,0,5.325,0,7.988 c-2.146-1.278-5.411,0.577-6.269,2.683c-1.074,2.636,1.925,3.462,3.907,2.736c1.954-0.715,3.253-2.269,3.343-4.333 c0.046-1.037,0-2.087,0-3.125c0-1.638,0-3.276,0-4.914c0-0.673,0-1.346,0-2.019c0-0.215-0.116-0.848,0-1.036 c0.17-0.277,1.196-0.348,1.535-0.427c1.425-0.329,2.849-0.657,4.274-0.986c2.579-0.595,5.158-1.19,7.737-1.786 c0,3.284,0,6.569,0,9.853c-1.64-0.977-3.949-0.013-5.2,1.179c-0.958,0.913-1.87,2.579-0.84,3.778 c1.198,1.393,3.567,0.71,4.866-0.135C10.164,6.578,10.936,5.347,11,3.788c0.148-3.605,0-7.245,0-10.853c0-0.415,0-0.829,0-1.244 C11-9.247,11.189-10.396,10.988-11.314z"
            fill="#fccc12"
            transform="translate(50,130)"
          >
            <animatetransform
              attributeName="transform"
              begin="0s"
              calcMode="spline"
              dur="1.7857142857142856s"
              keySplines="0 0 1 1;0 0.5 0.5 1;0.5 0 1 0.5;0 0 1 1"
              keyTimes="0;0;0.4;0.8;1"
              repeatCount="indefinite"
              type="translate"
              values="50 130;50 130;50 20;50 130;50 130;"
            />
          </path>
        </svg>
      </div>
    </div>
  </div>
</body>
`;
